internal/cli/custom_domains.go

```diff
@@ -0,0 +1,319 @@
+package cli
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+	"github.com/auth0/auth0-cli/internal/auth0"
+	"github.com/auth0/auth0-cli/internal/prompt"
+	"github.com/spf13/cobra"
+	"gopkg.in/auth0.v5/management"
+)
+
+const (
+	customDomainProvisioningTypeAuth0 = "auth0_managed_certs"
+	customDomainProvisioningTypeSelf = "self_managed_certs"
+	customDomainVerificationMethodTxt = "txt"
+)
+
+var (
+	customDomainID = Argument{
+		Name: "Id",
+		Help: "Id of the custom domain.",
+	}
+
+	customDomainDomain = Flag{
+		Name:       "Domain",
+		LongForm:   "domain",
+		ShortForm:  "d",
+		Help:       "Domain name.",
+		IsRequired: true,
+	}
+
+	customDomainType = Flag{
+		Name:       "Provisioning Type",
+		LongForm:   "type",
+		ShortForm:  "t",
+		Help:       "Custom domain provisioning type. Must be 'auth0' for Auth0-managed certs or 'self' for self-managed certs.",
+	}
+
+	customDomainVerification = Flag{
+		Name:      "Verification Method",
+		LongForm:  "verification",
+		ShortForm: "v",
+		Help:      "Custom domain verification method. Must be 'txt'.",
+	}
+)
+
+func customDomainsCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "domains",
+		Short: "Manage custom domains",
+		Long:  "Manage custom domains.",
+	}
+
+	cmd.SetUsageTemplate(resourceUsageTemplate())
+	cmd.AddCommand(listCustomDomainsCmd(cli))
+	cmd.AddCommand(showCustomDomainCmd(cli))
+	cmd.AddCommand(createCustomDomainCmd(cli))
+	cmd.AddCommand(deleteCustomDomainCmd(cli))
+	cmd.AddCommand(verifyCustomDomainCmd(cli))
+
+	return cmd
+}
+
+func listCustomDomainsCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		Short:   "List your custom domains",
+		Long: `List your existing custom domains. To create one try:
+auth0 branding domains create`,
+		Example: `auth0 branding domains list
+auth0 branding domains ls`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var list []*management.CustomDomain
+
+			if err := ansi.Waiting(func() error {
+				var err error
+				list, err = cli.api.CustomDomain.List()
+				return err
+			}); err != nil {
+				return fmt.Errorf("An unexpected error occurred: %w", err)
```

Comment:
shouldn't use https://github.com/auth0/auth0-cli/blob/77c96d394def391a2ed90d94485634230140b955/internal/auth0/error.go#L5 (pkg/errors under the hood)?

Reply:
We're just using that method for panics for now.

Reply:
(so that Sentry has a stack trace when it recovers from the panic)